frontend/src/components/invoice/invoice-edit/EditInvoice.tsx

```diff
@@ -79,6 +105,39 @@ const EditInvoice = () => {
   // --> There should be a form variant and a model variant new'd for every render
   // --> problem right now is that ex for a new invoice the invoice._id is not set after save
   // --> the invoice.attachments are also not updated because they are separate from the form...
+  const syncCreditNotas = (invoice: InvoiceModel) => {
+    const removedCreditNota = initInvoice.creditNotas.filter(n => !invoice.creditNotas.includes(n));
+    const creditNotaGroup = [...invoice.creditNotas, invoice.number]
+
+
+    removedCreditNota.forEach(creditnota => {
+      const invoiceToUpdate = invoices.find(i => i.number === creditnota);
+      if(invoiceToUpdate) {
+        invoiceToUpdate.creditNotas = []
+
+        dispatch(updateInvoiceRequest(invoiceToUpdate, undefined, false) as any);
+      }
+    })
+
+    invoice.creditNotas.forEach(creditNota => {
+      const invoiceToUpdate = invoices.find(i => i.number === creditNota);
+      if(invoiceToUpdate) {
+        const newCreditNotas = creditNotaGroup.filter(n => n !== invoiceToUpdate.number)
+
+        if (newCreditNotas.length !== invoiceToUpdate.creditNotas.length ||
+          !(newCreditNotas.every(num => invoiceToUpdate.creditNotas.includes(num)) &&
+          invoiceToUpdate.creditNotas.every(num => newCreditNotas.includes(num)))
+        ) {
+          invoiceToUpdate.creditNotas = newCreditNotas
+          dispatch(updateInvoiceRequest(invoiceToUpdate, undefined, false) as any);
+        }
+
+
+      }
+    })
+
+  }
```

Comment:
Is het mogelijk om deze code te doen in de updateInvoiceRequest() in plaats van in dit component?